### PR TITLE
feat: Add support for initializing custom context object

### PR DESCRIPTION
**Pull request by salmanm**
Fixes https://github.com/mcollina/autocannon/issues/279

- Added `initialContext` option which is used to initialize context instead of empty object
- Covered with tests
- Updated readme.

### CODE REVIEW

File: lib/requestIterator.js [hunk 1]
```diff
@@ -3,27 +3,33 @@
...
+  this.resetContext()
   this.reqDefaults = opts
   this.requestBuilder = requestBuilder(opts)
   this.setRequests(opts.requests)
 }
 
 inherits(RequestIterator, Object)
 
+RequestIterator.prototype.resetContext = function () {
+  this.context = clone(this.initialContext)
```
Comment (mcollina) [inline lib/requestIterator.js:24]: please use the `rdfc` module instead.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: lib/requestIterator.js (hunk 1 above)
Comment (salmanm) [reply lib/requestIterator.js:24]: Aye, done. Also replaced another pre-existing use of `clone`.

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/requestIterator.js (hunk 1 above)
Comment (salmanm) [reply lib/requestIterator.js:24]: Turns out, `rfdc` doesn't support copying `Buffer`s. I'll take that separately. Reverted the last commit.

**Comment by mcollina** [comment]
can you fix the conflicts?

**Pull request merged**